src/scene_battle_rpg2k3.cpp

```diff
@@ -352,10 +352,6 @@ void Scene_Battle_Rpg2k3::ResetAllBattlerZ() {
 			sprite->ResetZ();
 			sprite->DetectStateChange();
 		}
-		auto* weapon = actor->GetWeaponSprite();
-		if (weapon) {
-			weapon->ResetZ();
-		}
```

Comment:
uhm these 4 lines can stay. I do not think they are wrong and there is a ResetZ in the sprite class, so it will "just work".

Reply:
Well, these lines are no longer needed because `ResetZ` has been removed from `Sprite_Weapon`.

Reply:
Yes but ResetZ is still on Sprite_Battler and weapon inherits from it so you can still call it. But if this does not cause any issues it can stay removed.

Reply:
For me it works, because the Z-value gets set explicitly when `StartAttack` is called which makes the weapon visible until `StopAttack` is called. Outside from normal attacks the weapon is invisible so the Z-value does not matter then.